python/lsst/afw/detection/_footprintContinued.py

```diff
@@ -0,0 +1,68 @@
+# This file is part of afw.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from lsst.afw.image import Image
+
+import numpy as np
+
+from lsst.utils import continueClass
+
+from lsst.geom import Point2I
+from ._detection import Footprint
+
+
+__all__ = []  # import this module only for its side effects
+
+
+@continueClass
+class Footprint:  # noqa: F811
+    def extractFluxFromArray(self, image: np.ndarray, xy0: Point2I) -> float:
+        """
+
+        Parameters
+        ----------
+        image:
+            The array containing the image pixels to extract
+        xy0:
+            The origin of the image array.
+
+        Returns
+        flux:
+            The flux from the image in pixels contained in the footprint
+        """
+        return self.spans.flatten(image, xy0).sum()
+
+    def extractFluxFromImage(self, image: "Image") -> float:
```

Comment:
I don't know a lot about mypy, but I think this?

```suggestion
    def extractFluxFromImage(self, image: lsst.afw.image.Image) -> float:
```

Reply:
This is a problem with a circular dependency that is introduced by type checking. See https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/ .